src/OpenTelemetry.Instrumentation.ConfluentKafka/ConfluentKafkaMeterInstrumentation.cs

```diff
@@ -0,0 +1,194 @@
+        this.MessageSize = this.meter.CreateObservableGauge(Gauges.MessageSize, this.GetMessageSizeMeasurements, Descriptions[Gauges.MessageSize]);
+        this.Tx = this.meter.CreateCounter<long>(Counters.Tx, Descriptions[Counters.Tx]);
+        this.TxBytes = this.meter.CreateCounter<long>(Counters.TxBytes, Descriptions[Counters.TxBytes]);
+        this.TxMessages = this.meter.CreateCounter<long>(Counters.TxMessages, Descriptions[Counters.TxMessages]);
+        this.TxMessageBytes = this.meter.CreateCounter<long>(Counters.TxMessageBytes, Descriptions[Counters.TxMessageBytes]);
+        this.Rx = this.meter.CreateCounter<long>(Counters.Rx, Descriptions[Counters.Rx]);
+        this.RxBytes = this.meter.CreateCounter<long>(Counters.RxBytes, Descriptions[Counters.RxBytes]);
+        this.RxMessages = this.meter.CreateCounter<long>(Counters.RxMessages, Descriptions[Counters.RxMessages]);
+        this.RxMessageBytes = this.meter.CreateCounter<long>(Counters.RxMessageBytes, Descriptions[Counters.RxMessageBytes]);
+    }
+
+    public ObservableGauge<long> ReplyQueue { get; }
+
+    public ObservableGauge<long> MessageCount { get; }
+
+    public ObservableGauge<long> MessageSize { get; }
+
+    public Counter<long> Tx { get; }
+
+    public Counter<long> TxBytes { get; }
+
+    public Counter<long> TxMessages { get; }
+
+    public Counter<long> TxMessageBytes { get; }
+
+    public Counter<long> Rx { get; }
+
+    public Counter<long> RxBytes { get; }
+
+    public Counter<long> RxMessages { get; }
+
+    public Counter<long> RxMessageBytes { get; }
+
+    public ConcurrentQueue<Measurement<long>> ReplyQueueMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+
+    public ConcurrentQueue<Measurement<long>> MessageCountMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+
+    public ConcurrentQueue<Measurement<long>> MessageSizeMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+
+    public async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (await this.channel.Reader.WaitToReadAsync(stoppingToken).ConfigureAwait(false))
+        {
+            while (this.channel.Reader.TryRead(out var json))
+            {
+                Statistics? statistics;
+                try
+                {
+                    statistics = JsonSerializer.Deserialize(json, StatisticsJsonSerializerContext.Default.Statistics);
+                }
+                catch
+                {
+                    return;
+                }
+
+                if (statistics == null || statistics.Name == null)
+                {
+                    return;
+                }
+
+                TagList tags = new()
+                {
+                    { Tags.ClientId, statistics.ClientId },
+                    { Tags.Name, statistics.Name },
+                };
```

Comment:
To be in conformance with [messaging semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-metrics.md), there should be a tag for `messaging.system`, and for `messaging.destination.name` and `server.address` if those are available.